compliance_controller/compliance_controller/src/compliance_controller.cpp

```diff
@@ -0,0 +1,315 @@
+      JointTrajectoryController::desired_state_.velocity[i] = JointTrajectoryController::desired_joint_state_.velocity[0];
+      JointTrajectoryController::desired_state_.acceleration[i] = JointTrajectoryController::desired_joint_state_.acceleration[0]; ;
+
+      JointTrajectoryController::state_joint_error_.position[0] = angles::shortest_angular_distance(JointTrajectoryController::current_state_.position[i], 
+        JointTrajectoryController::desired_joint_state_.position[0]);
+      JointTrajectoryController::state_joint_error_.velocity[0] = JointTrajectoryController::desired_joint_state_.velocity[0] - JointTrajectoryController::current_state_.velocity[i];
+      JointTrajectoryController::state_joint_error_.acceleration[0] = 0.0;
+
+      JointTrajectoryController::state_error_.position[i] = angles::shortest_angular_distance(JointTrajectoryController::current_state_.position[i],
+        JointTrajectoryController::desired_joint_state_.position[0]);
+      JointTrajectoryController::state_error_.velocity[i] = JointTrajectoryController::desired_joint_state_.velocity[0] - JointTrajectoryController::current_state_.velocity[i];
+      JointTrajectoryController::state_error_.acceleration[i] = 0.0;
+
+      //Check tolerances
+      const typename JointTrajectoryController::RealtimeGoalHandlePtr rt_segment_goal = segment_it->getGoalHandle();
+      if (rt_segment_goal && rt_segment_goal == JointTrajectoryController::rt_active_goal_)
+      {
+        // Check tolerances
+        if (time_data.uptime.toSec() < segment_it->endTime())
+        {
+          // Currently executing a segment: check path tolerances
+          const joint_trajectory_controller::SegmentTolerancesPerJoint< Scalar >& joint_tolerances = segment_it->getTolerances();
+
+          if (!checkStateTolerancePerJoint(JointTrajectoryController::state_joint_error_, joint_tolerances.state_tolerance))
+          {
+            if (JointTrajectoryController::verbose_)
+            {
+              ROS_ERROR_STREAM_NAMED(JointTrajectoryController::name_,"Path tolerances failed for joint: " << JointTrajectoryController::joint_names_[i]);
+              checkStateTolerancePerJoint(JointTrajectoryController::state_joint_error_, joint_tolerances.state_tolerance, true);
+            }
+            rt_segment_goal->preallocated_result_->error_code =
+              control_msgs::FollowJointTrajectoryResult::PATH_TOLERANCE_VIOLATED;
+            rt_segment_goal->setAborted(rt_segment_goal->preallocated_result_);
+            JointTrajectoryController::rt_active_goal_.reset();
+            JointTrajectoryController::successful_joint_traj_.reset();
+          }
+        }
+        else if (segment_it == --curr_traj[i].end())
+        {
+          if (JointTrajectoryController::verbose_)
+            ROS_DEBUG_STREAM_THROTTLE_NAMED(1,JointTrajectoryController::name_,"Finished executing last segment, checking goal tolerances");
```

Comment:
it would be nice to run clang-format on this PR, for example missing spaces between arguments